### PR TITLE
[BEAM-2595] Allow table schema objects in BQ DoFn

### PR DESCRIPTION
**Pull request by sb2nov**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

R: @chamikaramj PTAL

### CODE REVIEW

**Comment by sb2nov** [comment]
Retest this please

File: sdks/python/apache_beam/io/gcp/bigquery_test.py [hunk 1]
```diff
@@ -995,6 +1021,45 @@ def test_simple_schema_parsing(self):
         fields=[string_field, number_field])
     self.assertEqual(expected_table_schema, table_schema)
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcp/bigquery_test.py]: Pls expand this test to convert this schema to a dict and compare with the original dict. Also include nested fields.

File: sdks/python/apache_beam/io/gcp/bigquery.py [hunk 2]
```diff
@@ -1192,21 +1192,12 @@ def __init__(self, table_id, dataset_id, project_id, batch_size, schema,
   @staticmethod
   def get_table_schema(schema):
```
Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: Done

File: sdks/python/apache_beam/io/gcp/bigquery.py [hunk 3]
```diff
@@ -1192,21 +1192,12 @@ def __init__(self, table_id, dataset_id, project_id, batch_size, schema,
...
-        field_schema = bigquery.TableFieldSchema()
-        field_schema.name = field_name
-        field_schema.type = field_type
-        field_schema.mode = 'NULLABLE'
-        table_schema.fields.append(field_schema)
-      return table_schema
-    elif schema is None:
+    if schema is None:
       return schema
     elif isinstance(schema, bigquery.TableSchema):
```
Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: Done

File: sdks/python/apache_beam/io/gcp/bigquery.py [hunk 4]
```diff
@@ -1289,13 +1280,65 @@ def __init__(self, table, dataset=None, project=None, schema=None,
...
+      field_name, field_type = field_and_type.split(':')
+      field_schema = bigquery.TableFieldSchema()
+      field_schema.name = field_name
+      field_schema.type = field_type
+      field_schema.mode = 'NULLABLE'
+      table_schema.fields.append(field_schema)
+    return table_schema
+
+  @staticmethod
+  def table_schema_to_dict(table_schema):
```
Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: Done

File: sdks/python/apache_beam/io/gcp/bigquery.py [hunk 5]
```diff
@@ -1289,13 +1280,65 @@ def __init__(self, table, dataset=None, project=None, schema=None,
...
+
+    if not isinstance(table_schema, bigquery.TableSchema):
+      raise ValueError("Table schema must be of the type bigquery.TableSchema")
+    schema = {'fields': []}
+    for field in table_schema.fields:
+      schema['fields'].append(get_table_field(field))
+    return schema
+
+  @staticmethod
+  def get_dict_table_schema(schema):
```
Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: Done

File: sdks/python/apache_beam/io/gcp/bigquery_test.py (hunk 1 above)
Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/bigquery_test.py]: Updated with nested fields and added an end-2-end test.

**Comment by chamikaramj** [comment]
LGTM

**Pull request closed without merging**